### PR TITLE
Update release workflow docs

### PR DESCRIPTION
**Pull request by MarekMichali**
**Description**

Changes proposed in this pull request:

- Updates release diagram
- Updates release workflow docs

**Related issue(s)**

### CODE REVIEW

File: docs/contributor/03-10-release.md [hunk 1]
```diff
@@ -21,14 +21,19 @@ To create a release, follow these steps:
...
-5. `post-btp-manager-build` builds a Docker image tagged with the release name.
-6. `post-btp-manager-module-build` runs the `kyma alpha create module` command, which creates a Kyma module and pushes the image to the registry. Kyma CLI is called with the `--sec-scanners-config` flag and uses a dynamically created file to configure security scanning settings in the module template. Finally, the job uploads the `template.yaml`,`template_control_plane.yaml`, `btp-manager.yaml` and `btp-operator-default-cr.yaml` files to the btp-manager release as release assets.
-7. The GitHub action waits for the `template.yaml` asset in the GitHub release and for images in the Docker registry.
-8. The GitHub action fetches the module image and runs E2E tests on the k3s cluster with the specified credentials. 
-9. If the unit tests and E2E tests are completed successfully, the GitHub action publishes the release.
+   vi. choose to bump or not the security scanner config
```
Comment (IwonaLanger) [inline docs/contributor/03-10-release.md:24]:
```suggestion
   vi. choose whether to bump or not to bump the security scanner config
```

File: docs/contributor/03-10-release.md [hunk 2]
```diff
@@ -21,14 +21,19 @@ To create a release, follow these steps:
...
+2. The GitHub action, defined in the [`create-release.yaml`](/.github/workflows/create-release.yaml) file, validates the release by checking if the GitHub tag already exists, if there are any old docker images for that GitHub tag, and if any merged PR's that will be part of this release are labeled correctly.
+3. The GitHub action (if you chose to bump the security scanner config) will create a PR with a new security scanner config that includes the new GitHub tag version.
+4. A code owner approves the PR. 
+5. The GitHub action creates a GitHub tag and draft release with the provided name.
+6. The GitHub action asynchronously initiates unit tests and an await for Prow Jobs status.
```
[the remark this answers is not in the document]
Comment (MarekMichali) [reply docs/contributor/03-10-release.md:29]: Yes it initiates a step that will wait for the Prow Jobs status.

[the remark this answers is not in the document]
Comment (MarekMichali) [reply docs/contributor/03-10-release.md:29]: The `unit tests` and the `await for Prow Jobs status` are separate steps.

File: docs/contributor/03-10-release.md (hunk 1 above)
Comment (jaroslaw-pieszka) [inline docs/contributor/03-10-release.md:24]:
```suggestion
   vi. choose whether to bump or not to bump the security scanner config file
```

Comment (MarekMichali) [reply docs/contributor/03-10-release.md]: @IwonaLanger suggested the same change, but without the last word `file`. I think without the word it better describes what is happening. We are not bumping the file, we are bumping the "config". Or it's just me misunderstanding something.

File: docs/contributor/03-10-release.md [hunk 3]
```diff
@@ -21,14 +21,19 @@ To create a release, follow these steps:
...
-7. The GitHub action waits for the `template.yaml` asset in the GitHub release and for images in the Docker registry.
-8. The GitHub action fetches the module image and runs E2E tests on the k3s cluster with the specified credentials. 
-9. If the unit tests and E2E tests are completed successfully, the GitHub action publishes the release.
+   vi. choose whether to bump or not to bump the security scanner config
+2. The GitHub action, defined in the [`create-release.yaml`](/.github/workflows/create-release.yaml) file, validates the release by checking if the GitHub tag already exists, if there are any old Docker images for that GitHub tag, and if merged PR's that will be part of this release are labeled correctly.
```
Comment (IwonaLanger) [inline docs/contributor/03-10-release.md:25]:
```suggestion
2. The GitHub action, defined in the [`create-release.yaml`](/.github/workflows/create-release.yaml) file, validates the release by checking if the GitHub tag already exists, if there are any old Docker images for that GitHub tag, and if merged PRs that will be part of this release are labeled correctly.
```

**Review by IwonaLanger — APPROVED** [review]

**Review by jaroslaw-pieszka — APPROVED** [review]

**Pull request merged**